wandb/sdk/wandb_setup.py

```diff
@@ -177,6 +177,8 @@ def _load_viewer(self, settings=None):
         self._server = s
 
     def _load_user_settings(self, settings=None):
+        if self._settings and self._settings._offline:
```

Comment:
That's interesting that you need this condition. From reading the code it seems that we are not suppose to call this function if `settings` is `offline`:

- https://github.com/wandb/client/blob/master/wandb/sdk/wandb_init.py#L195-L196

- and here: https://github.com/wandb/client/blob/master/wandb/sdk/wandb_setup.py#L141
which is referenced by this one: https://github.com/wandb/client/blob/master/wandb/sdk/wandb_login.py#L174-L175

what am I missing?

Reply:
yea, you're absolutely right, but a user somehow managed to trigger this: https://community.wandb.ai/t/unable-to-run-wandb-online-after-running-offline/1252/2

Reply:
That's very strange, but I know that `Settings` still needs a bit of work, so it might be possible that we don't catch the correct settings. Were you able to trigger this case as well? In any case the change is harmless.

Reply:
Honestly, not quite sure where to test it - if self._settings._offline is True, this line should never be reached in the first place. I'd guess the user's setting/env got mingled somehow, but I'm having a hard time reproducing this.
Any suggestions?

Reply:
You were right! The user had a really mangled environment. This is the minimal script to recreate the original issue:

```python
import wandb
import os

os.environ["WANDB_MODE"] = "offline"

wandb.init(mode="online")
```
So it seems that the problem is our handling of settings. We don't respect the hierarchy, we basically set the env variable to settings and don't update the settings in with the `mode` argument.

Reply:
Nice, thank you! What should the priorities be? Defaults -> env variables -> user-supplied settings? cc @raubitsj?
I'd be happy to close this PR and instead work on enforcing the priorities.

Reply:
These suppose to be the priorities: https://github.com/wandb/client/blob/master/wandb/sdk/wandb_settings.py#L293-L306
It seems that for some reason we don't update the `mode` settings with the relevant arg that we pass to init... But I'll let @raubitsj give more details.